downstairs/src/lib.rs

```diff
@@ -1410,56 +1408,83 @@ impl ActiveConnection {
                 continue;
             };
 
-            cdt::work__process!(|| new_id.0);
-            let m = self.do_work(&dw, flags, reqwest_client, region).await;
-
-            if let Some(error) = m.err() {
-                self.reply(Message::ErrorReport {
-                    upstairs_id: upstairs_connection.upstairs_id,
-                    session_id: upstairs_connection.session_id,
-                    job_id: new_id,
-                    error: error.clone(),
-                })?;
+            if let Some(failed) =
+                self.do_work(dw, flags, reqwest_client, dss, region).await?
+            {
+                new_work.push_back(failed);
+            }
+        }
+        Ok(())
+    }
 
-                // If the job errored, do not consider it completed.
-                // Retry it.
-                new_work.push_back(new_id);
-                self.work.active.insert(dw.ds_id, dw);
+    /// Do work for the given IO job
+    ///
+    /// This function will either complete the work (adding it to the completed
+    /// list), fail to complete the work (re-adding it to the active list and
+    /// returning the `JobId`), or fail to communicate when replying to the
+    /// Upstairs (which returns an error).
```

Comment:
There should be an explicit mention of when this returns None, as well as a mention of when the entire task should bail due to an error during live repair.

Reply:
Yup, done in 872351db691c0f1debf1264adb77e1060741068b